content/spaces/locking.textile

```diff
@@ -0,0 +1,215 @@
+
+Each lock is identified by a unique string ID, and only a single member may hold a lock with a given string at any one time. A lock will exist in one of three "states":#states and may only transition between states in specific circumstances.
+
+h2(#states). Lock states
+
+Component locking is handled entirely client-side. Members will optimistically begin to edit a component as soon as they call "acquire()":#acquire on the lock identifier related to it. A short window will follow before it is confirmed whether the lock request was successful or rejected.
+
+A lock will be in one of the following states:
+
+- @pending@ := A member has requested a lock by calling "@acquire()@":#acquire.
+- @locked@ := The lock is confirmed to be held by the requesting member.
+- @unlocked@ := The lock is confirmed to not be locked by the requesting member, or has been "released":#release by a member previously holding the lock.
+
+The following lock state transitions may occur. Only transitions that result in a @locked@ or @unlocked@ status will emit a lock event that members can "@subscribe()@":#subscribe to.
+
+* None --> @pending@: a member calls "@acquire()@":#acquire to request a lock.
+* @pending@ --> @locked@: the requesting member holds the lock.
+* @pending@ --> @unlocked@: the requesting member does not hold the lock, since another member already holds it.
+* @locked@ --> @unlocked@: the lock was either explicitly "released":#release by the member, or their request was invalidated by a concurrent request which took precedence.
+* @unlocked@ --> @locked@: the requesting member reacquired a lock they previously held.
+
+h2(#acquire). Acquire a lock
+
+Use the @acquire()@ method to attempt to acquire a lock with a given unique ID. Additional @attributes@ may be passed when trying to acquire a lock that can contain a set of arbitrary key-value pairs. An example of using @attributes@ is to store the component ID the lock relates to so that it can be easily updated in the UI with a visual indication of its lock status.
+
+A member must have been "entered":/spaces/space#enter into the space to acquire a lock.
+
+The following is an example of attempting to acquire a lock:
+
+```[javascript]
+const acquireLock = await space.locks.acquire(id);
+```
+
+The following is an example of passing a set of @attributes@ when trying to acquire a lock:
+
+```[javascript]
+const lockAttributes = new Map();
+lockAttributes.set('component', 'cell-d3');
+const acquireLock = await space.locks.acquire(id, { lockAttributes });
+```
+
+Once a member requests a lock by calling @acquire()@, the lock is temporarily in the "pending state":#states. An event will be emitted based on whether the lock request was successful (a status of @locked@) or invalidated (a status of @unlocked@).
```

Comment:
We have noted what the status of the lock would be but what will be returned and saved in the variable `aquireLock` in the above snippet?